core/src/main/java/hudson/model/AbstractItem.java

```diff
@@ -228,6 +233,102 @@ protected void doSetName(String name) {
         this.name = name;
     }
 
+    /**
+     * @return whether {@link #name} can be modified by a user
+     */
+    public boolean isNameEditable() {
```

Comment:
`canRename` might have been a more intuitive name, but I see that you are pulling up the existing `Job` method, so too late to change.